gitlab/resource_gitlab_instance_cluster.go

```diff
@@ -0,0 +1,248 @@
+				Optional:     true,
+				ForceNew:     true,
+				Default:      "rbac",
+				ValidateFunc: validation.StringInSlice([]string{"rbac", "abac", "unknown_authorization"}, false),
+			},
+			"management_project_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceGitlabInstanceClusterCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gitlab.Client)
+
+	pk := gitlab.AddPlatformKubernetesOptions{
+		APIURL: gitlab.String(d.Get("kubernetes_api_url").(string)),
+		Token:  gitlab.String(d.Get("kubernetes_token").(string)),
+	}
+
+	if v, ok := d.GetOk("kubernetes_ca_cert"); ok {
+		pk.CaCert = gitlab.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("kubernetes_authorization_type"); ok {
+		pk.AuthorizationType = gitlab.String(v.(string))
+	}
+
+	options := &gitlab.AddClusterOptions{
+		Name:               gitlab.String(d.Get("name").(string)),
+		Enabled:            gitlab.Bool(d.Get("enabled").(bool)),
+		Managed:            gitlab.Bool(d.Get("managed").(bool)),
+		PlatformKubernetes: &pk,
+	}
+
+	if v, ok := d.GetOk("domain"); ok {
+		options.Domain = gitlab.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("environment_scope"); ok {
+		options.EnvironmentScope = gitlab.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("management_project_id"); ok {
+		options.ManagementProjectID = gitlab.String(v.(string))
+	}
+
+	log.Printf("[DEBUG] create gitlab instance cluster %q", *options.Name)
+
+	cluster, _, err := client.InstanceCluster.AddCluster(options)
+
+	if err != nil {
+		return err
+	}
+
+	clusterIdString := fmt.Sprintf("%d", cluster.ID)
+	d.SetId(clusterIdString)
+
+	return resourceGitlabInstanceClusterRead(d, meta)
+}
+
+func resourceGitlabInstanceClusterRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gitlab.Client)
+
+	clusterId, err := strconv.Atoi(d.Id())
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] read gitlab instance cluster %d", clusterId)
+
+	cluster, _, err := client.InstanceCluster.GetCluster(clusterId)
+	if err != nil {
+		return err
+	}
+
+	d.Set("name", cluster.Name)
+	d.Set("domain", cluster.Domain)
+	d.Set("created_at", cluster.CreatedAt.String())
+	d.Set("provider_type", cluster.ProviderType)
+	d.Set("platform_type", cluster.PlatformType)
+	d.Set("environment_scope", cluster.EnvironmentScope)
+	d.Set("cluster_type", cluster.ClusterType)
+
+	d.Set("kubernetes_api_url", cluster.PlatformKubernetes.APIURL)
+	d.Set("kubernetes_ca_cert", cluster.PlatformKubernetes.CaCert)
+	d.Set("kubernetes_namespace", cluster.PlatformKubernetes.Namespace)
+	d.Set("kubernetes_authorization_type", cluster.PlatformKubernetes.AuthorizationType)
+
+	if cluster.ManagementProject == nil {
+		d.Set("management_project_id", "")
+	} else {
+		d.Set("management_project_id", strconv.Itoa(cluster.ManagementProject.ID))
+	}
+
+	return nil
+}
+
+func resourceGitlabInstanceClusterUpdate(d *schema.ResourceData, meta interface{}) error {
```

Comment:
Looks like you're missing setting the `enabled` and `kubernetes_namespace` attributes in `EditClusterOptions` in this function.

Reply:
Thanks! Looks like `enabled` wasn't an `EditClusterOptions` attribute, made a PR to add it: https://github.com/xanzy/go-gitlab/pull/939

Reply:
Cool. In the meantime we can omit the `enabled` attribute from the schema or make it `ForceNew: true`.